engine/src/main/java/org/terasology/rendering/logic/FloatingTextRenderer.java

```diff
@@ -99,6 +91,14 @@ private void render(Iterable<EntityRef> floatingTextEntities) {
 
             FloatingTextComponent floatingText = entity.getComponent(FloatingTextComponent.class);
 
+            // If the FloatingTextComponent is meant to be occluded, enable depth tests
+            if (floatingText.isOccluded) {
+                glEnable(GL_DEPTH_TEST);
+                glDepthFunc(GL_LESS);
+            } else {
+                glDisable(GL_DEPTH_TEST);
+            }
+
```

Comment:
Hmm.. you are enabling it, but where are you disabling it back?
Always try to revert system back to the state you found it in.

Reply:
Depth testing, by default, is enabled, otherwise _everything_ would be rendered on the screen. The method in its _previous_ and current state enables depth testing before it returns.

Reply:
> Depth testing, by default, is enabled, otherwise everything would be rendered on the screen.
Depth testing in OpenGL is not enabled by default.

That being said, you are correct, it does have a line to disable it back - it didn't appear in the diff, because it was untouched by you.

Reply:
I meant that in the context of the Terasology rendering engine, where depth testing enabled should be the "default" state.

Reply:
Depth Testing is indeed enabled by default in LWJGLGraphics.initOpenGLParams(): it is a Terasology default and is the opposite of the OpenGL default. 

That doesn't mean that's the way it "should" be, as only a limited number of nodes in the renderer actually requires depth testing enabled. Properly evaluating this aspect is not in the scope of this discussion albeit it is one of the things I'd like to do at some point.

More relevant is the way depth testing is enabled and disabled in this method. 

Right now depth testing is enabled or disabled in the first part of the method, within the outer for loop, and then "universally" re-enabled outside of it. I have a few comments:

1. rely on the default (depth testing enabled) and do nothing if the default is enough.
2. if depth testing must be disabled, re-enable it on a per-entity basis (two IF blocks)
2b. if you wanted to be really fancy, sort the entities first, to do the least enabling/disabling
3. move the code to enable/disable things as close as possible to the rendering code, i.e. just outside or inside the two glPush/PopMatrix statements.
4. If glDepthFunc(GL_LESS) is really needed (Terasology default is GL_LEQUAL) that can be set and reset to GL_LEQUAL outside of the for(entity) loop, just once. 

That been said, isn't LEQUAL enough?

Reply:
Oh. I just noticed the removal of the glDepthFunc(GL_LESS) line in the next commit. Never mind point 4 then!

Reply:
Sorting will be overkill, but I agree with the other two points. Changes incorporated.